### ISSUE TITLE
Pulse schedule of the default CNOT gate does not give the correct behavior

### ISSUE THREAD

**Issue by alejomonbar** [Qiskit/qiskit-aer#1093]
### Informations

- **Qiskit Aer version**: 0.7.1
- **Python version**: 3.8.3
- **Operating system**: macOS

### What is the current behavior?
When transpiling and using the schedule from the default CNOT gate of the fake Valencia device the result does not agree with the result of the CNOT gate.

### Steps to reproduce the problem

```  python
from qiskit import transpile, schedule
from qiskit.circuit import QuantumRegister, QuantumCircuit, Gate
from qiskit.pulse import Schedule, Play, ControlChannel
from qiskit.pulse.library import GaussianSquare
from qiskit.ignis.verification.tomography import process_tomography_circuits
from qiskit.test.mock import FakeAlmaden, FakeValencia
import qiskit as qk
import matplotlib.pyplot as plt
from qiskit.visualization import SchedStyle
plt.rcParams['figure.figsize'] = [5, 5]
style = SchedStyle(figsize=(20, 10))


backend = FakeValencia()

qc = QuantumCircuit(2,2, name = "CNOT")
qc.cx(0,1)
qc.measure(range(2),range(2))
qc = transpile(qc, backend)
pulse_schedule = schedule(qc, backend)

from qiskit.providers.aer.pulse import PulseSystemModel
from qiskit.compiler import assemble
from qiskit.providers.aer import PulseSimulator

backend_sim = PulseSimulator()
valencia_model = PulseSystemModel.from_backend(backend)
f = backend.properties().frequency
freq = [f(0), f(1), f(2), f(3), f(4)]
for uclo in valencia_model.u_channel_lo: 
    uclo[0].scale = complex(*uclo[0].scale)

shots = 512
CNOT_gate = assemble(pulse_schedule,
                    backend=backend_sim,
                    qubit_lo_freq=freq,
                    meas_level=2,
                    meas_return='single',
                    shots=shots)
sim_result = backend_sim.run(CNOT_gate , valencia_model).result()

cx_counts = sim_result.get_counts()

from qiskit.visualization import plot_histogram
plot_histogram(cx_counts)
```
### What is the expected behavior?
I should get a state closed to {'00000':512} but I'm getting the state {'00000':51, '00010':461}

**Comment by vvilpas** [comment]
Hi @alejomonbar! Thanks for reporting this. I've been able to reproduce with the version reported. However, using the latest `master` I get the expected result `{'00000':512}`. It seems it has been fixed in the PR that starts using the terra operators (#1014). This will be included in 0.8 release.

**Comment by vvilpas** [comment]
I'm closing this. Feel free to reopen if next release doesn't solve the problem.
